File: card.rb
class Card
  include Comparable

  attr_reader :rank, :suit

  def initialize(rank, suit)
    @rank, @suit = rank, suit
  end

  def <=>(other_card)
    self.value <=> other_card.value
  end

  def value
    case rank
      when 'Ace' then 14
      when 'King' then 13
      when 'Queen' then 12
      when 'Jack' then 11
      else; rank.to_i
    end
  end

  def face_card?
    value > 10 && value < 14
  end

  def ace?
    rank == 'Ace'
  end

  def red?
    [:diamonds, :hearts].include? suit.downcase.to_sym
  end

  def to_s
    "#{@rank} of #{@suit}"
  end

end

File: public/js/card.js
(function() {
  var Card;

  Card = (function() {
    function Card(rank, suit) {
      this.rank = rank;
      this.suit = suit;
    }

    Card.prototype.value = function() {
      switch (this.rank) {
        case "A":
          return 14;
        case "K":
          return 13;
        case "Q":
          return 12;
        case "J":
          return 11;
        default:
          return this.rank;
      }
    };

    Card.prototype.isAce = function() {
      return this.rank === 'A';
    };

    Card.prototype.isFaceCard = function() {
      return this.value > 10 && this.value < 14;
    };

    Card.prototype.color = function() {
      var _ref;

      if ((_ref = this.suit) === "Diamonds" || _ref === "Hearts") {
        return "Red";
      } else {
        return "Black";
      }
    };

    Card.prototype.toString = function() {
      return "" + (this.rankName()) + " of " + this.suit;
    };

    Card.prototype.rankName = function() {
      switch (this.rank) {
        case "A":
          return "Ace";
        case "K":
          return "K";
        case "Q":
          return "Queen";
        case "J":
          return "Jack";
        default:
          return this.rank;
      }
    };

    Card.prototype.suitSymbol = function() {
      switch (this.suit) {
        case "Diamonds":
          return '&diams;';
        case "Hearts":
          return '&hearts;';
        case "Spades":
          return '&spades;';
        case "Clubs":
          return '&clubs;';
        default:
          return '&#9800;';
      }
    };

    Card.prototype.suitClass = function() {
      return this.suit.toLowerCase();
    };

    return Card;

  })();

  exports.Card = Card;

}).call(this);
